File: backend/src/main/java/com/houserent/entity/House.java
package com.houserent.entity;

import javax.persistence.*;
import lombok.Data;
import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
@Entity
@Table(name = "houses")
public class House {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "landlord_id", nullable = false)
    private User landlord;

    @Column(nullable = false)
    private String title;

    @Column(columnDefinition = "TEXT")
    private String description;

    @Column(nullable = false)
    private String province;

    @Column(nullable = false)
    private String city;

    @Column(nullable = false)
    private String district;

    private String address;

    @Column(nullable = false)
    private BigDecimal price;

    @Column(nullable = false)
    private Integer area;

    @Column(nullable = false)
    private Integer bedroom;

    private Integer livingRoom;
    private Integer bathroom;

    private String orientation;
    private String floor;
    private String decoration;
    private String houseType;

    private String images;
    private String facilities;

    @Column(nullable = false)
    private String status;

    @Column(nullable = false)
    private LocalDateTime createdAt;

    private LocalDateTime updatedAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
        if (status == null) {
            status = "PUBLISHED";
        }
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }
}

File: frontend/src/app/models/house.ts
export interface House {
  id: number;
  landlord: {
    id: number;
    username: string;
    realName?: string;
    phone?: string;
  };
  title: string;
  description?: string;
  province: string;
  city: string;
  district: string;
  address?: string;
  price: number;
  area: number;
  bedroom: number;
  livingRoom?: number;
  bathroom?: number;
  orientation?: string;
  floor?: string;
  decoration?: string;
  houseType?: string;
  images?: string;
  facilities?: string;
  status: string;
  createdAt: string;
}

export interface HouseRequest {
  title: string;
  description?: string;
  province: string;
  city: string;
  district: string;
  address?: string;
  price: number;
  area: number;
  bedroom: number;
  livingRoom?: number;
  bathroom?: number;
  orientation?: string;
  floor?: string;
  decoration?: string;
  houseType?: string;
  images?: string;
  facilities?: string;
}

export interface HouseFilter {
  keyword?: string;
  city?: string;
  minPrice?: number;
  maxPrice?: number;
  minArea?: number;
  maxArea?: number;
  bedroom?: number;
  houseType?: string;
  decoration?: string;
}
